zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -737,6 +737,16 @@ impl Chain {
             .collect()
     }
 
+    /// Returns the Sapling [`NoteCommitmentSubtree`] if it was completed at the tip height.
+    pub fn sapling_subtree_for_tip(&self) -> Option<NoteCommitmentSubtree<sapling::tree::Node>> {
+        if !self.is_empty() {
+            let tip = self.non_finalized_tip_height();
+            Some(self.sapling_subtree(tip.into())?)
```

Comment:
Optional: Also can be done for Orchard.

```suggestion
            self.sapling_subtree(tip.into())
```